openmeter/app/httpdriver/customer.go

```diff
@@ -271,63 +273,79 @@ func (h *handler) getCustomerData(ctx context.Context, namespace string, apiApp
 			return nil, nil, fmt.Errorf("error converting to stripe app: %w", err)
 		}
 
-		// Get app ID from API data or get default app
-		app, err := h.getApp(ctx, namespace, apiStripeCustomerData.Id, app.AppTypeStripe)
+		// Resolve app
+		resolvedApp, err := h.resolveCustomerApp(ctx, customerID, app.AppTypeStripe, apiStripeCustomerData.Id)
 		if err != nil {
-			return nil, nil, fmt.Errorf("error getting stripe app: %w", err)
+			return nil, nil, fmt.Errorf("error resolving stripe app: %w", err)
 		}
 
+		// Create customer data
 		stripeCustomerData := appstripeentity.CustomerData{
 			StripeCustomerID:             apiStripeCustomerData.StripeCustomerId,
 			StripeDefaultPaymentMethodID: apiStripeCustomerData.StripeDefaultPaymentMethodId,
 		}
 
-		return app, stripeCustomerData, nil
+		return resolvedApp, stripeCustomerData, nil
 	case string(app.AppTypeCustomInvoicing):
 		// Parse as custom invoicing app
 		apiCustomInvoicingCustomerData, err := apiApp.AsCustomInvoicingCustomerAppData()
 		if err != nil {
 			return nil, nil, fmt.Errorf("error converting to custom invoicing app: %w", err)
 		}
 
-		// Get app ID from API data or get default app
-		app, err := h.getApp(ctx, namespace, apiCustomInvoicingCustomerData.Id, app.AppTypeCustomInvoicing)
+		// Resolve app
+		resolvedApp, err := h.resolveCustomerApp(ctx, customerID, app.AppTypeCustomInvoicing, apiCustomInvoicingCustomerData.Id)
```

Comment:
Earlier this resolved to the default Stripe app which was always the first one installed except someone didn't change it explicitly via API.